buildSrc/src/main/java/com/soundcloud/reflect/DaggerReflectPlugin.kt

```diff
@@ -1,20 +1,36 @@
 package com.soundcloud.reflect
 
+import org.gradle.api.GradleException
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.kotlin.dsl.dependencies
 import org.gradle.util.VersionNumber
 
 class DaggerReflectPlugin : Plugin<Project> {
     override fun apply(target: Project) {
+        if (target != target.rootProject) {
+            throw GradleException("Plugin must be applied to the root project")
+        }
+
         val extension = target.extensions.create("delect", DelectExtension::class.java)
 
-        if (!shouldActivateDaggerReflect(target)) {
-            return
+        target.afterEvaluate {
```

Comment:
I wonder if this `target.afterEvaluate` could be bad for performance if we're using dependency substition in here.

Reply:
If this runs only on the root project (which it does), I doubt that this will have a measurable impact. Do you think so?